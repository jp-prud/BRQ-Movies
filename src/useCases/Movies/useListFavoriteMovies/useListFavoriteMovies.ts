import { MoviesService } from '@services';
import { useQuery } from '@tanstack/react-query';
import { QueryKeys } from '@types';

export function useListFavoriteMovies() {
  const { listFavoriteMovies } = MoviesService();

  const { data, isLoading, isError } = useQuery({
    queryKey: [QueryKeys.listFavoriteMovies],
    queryFn: listFavoriteMovies,
  });

  return {
    favoriteMoviesData: {
      list: data,
      isLoading,
      isError,
    },
  };
}
