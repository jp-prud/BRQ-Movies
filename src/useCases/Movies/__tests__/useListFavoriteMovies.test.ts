import { renderHook, waitFor } from "@tests";
import { useListFavoriteMovies } from "../useListFavoriteMovies/useListFavoriteMovies";

describe('useListFavoriteMovies', () => { 
  it('should list favorite movies', async () => { 
    const { result } = renderHook(() => useListFavoriteMovies());
    
    await waitFor(() => expect(result.current.favoriteMoviesData.isLoading).toBe(false));

    expect(result.current.favoriteMoviesData.isLoading).toBe(false);
    expect(result.current.favoriteMoviesData).toMatchSnapshot()
  });
});