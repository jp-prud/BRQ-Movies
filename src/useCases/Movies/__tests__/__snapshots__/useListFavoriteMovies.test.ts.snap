// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useListFavoriteMovies should list favorite movies 1`] = `
{
  "isError": false,
  "isLoading": false,
  "list": null,
}
`;
